### PR TITLE
Update comments onboarding tour

### LINKED ISSUE #5577: [Task] Onboarding changes with the new bot flow for viewer and editor

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description
Comments tour updates

Fixes #5577

Depends on changes done in #5521 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: update-comments-onboarding-tour 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 53.78 **(0.03)** | 35.74 **(-0.03)** | 32.34 **(0.02)** | 54.36 **(0.04)**
 :red_circle: | app/client/src/api/OrgApi.ts | 56.25 **(-3.13)** | 0 **(0)** | 8.33 **(-8.34)** | 54.84 **(-3.22)**
 :red_circle: | app/client/src/comments/CommentCard/CommentCard.tsx | 72.67 **(-0.82)** | 63.25 **(-2.24)** | 42.86 **(-1.04)** | 73.13 **(-0.9)**
 :green_circle: | app/client/src/comments/inlineComments/AddCommentInput.tsx | 74.34 **(1.96)** | 42.5 **(0)** | 65 **(1.84)** | 74.75 **(2.22)**
 :green_circle: | app/client/src/comments/inlineComments/OverlayCommentsWrapper.tsx | 96.77 **(0.22)** | 75 **(0)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/comments/inlineComments/useOrgUsers.tsx | 95.45 **(-4.55)** | 77.78 **(2.78)** | 100 **(0)** | 95.45 **(-4.55)**
 :red_circle: | app/client/src/components/ads/tour/TourTooltipWrapper.tsx | 96.77 **(0)** | 69.44 **(0.69)** | 80 **(-3.33)** | 96.77 **(0)**
 :green_circle: | app/client/src/pages/Applications/permissionHelpers.tsx | 100 **(7.69)** | 100 **(0)** | 100 **(50)** | 100 **(8.33)**
 :red_circle: | app/client/src/pages/Editor/ToggleModeButton.tsx | 65.19 **(-5.08)** | 25.4 **(-13.24)** | 41.38 **(2.92)** | 64 **(-4.32)**
 :red_circle: | app/client/src/reducers/uiReducers/orgReducer.ts | 13.51 **(-2.71)** | 0 **(0)** | 0 **(-4.17)** | 13.51 **(-2.71)**
 :red_circle: | app/client/src/reducers/uiReducers/commentsReducer/handleNewCommentThreadEvent.ts | 86.67 **(-5.64)** | 75 **(-8.33)** | 100 **(0)** | 86.67 **(-5.64)**
 :red_circle: | app/client/src/sagas/OrgSagas.ts | 18.05 **(-4.51)** | 2.7 **(-0.33)** | 7.14 **(-7.15)** | 22.64 **(-3.78)**
 :green_circle: | app/client/src/selectors/entitiesSelector.ts | 48.02 **(0.25)** | 5.56 **(0)** | 19.05 **(-0.23)** | 45.78 **(0.33)**
 :green_circle: | app/client/src/selectors/organizationSelectors.tsx | 69.77 **(2.33)** | 0 **(0)** | 31.58 **(5.26)** | 66.67 **(3.71)**
 :red_circle: | app/client/src/utils/storage.ts | 37.23 **(-2.19)** | 13.64 **(0)** | 16.67 **(-3.33)** | 30 **(-2)**
 :sparkles: :new: | **app/client/src/utils/hooks/useOrg.tsx** | **88.24** | **27.27** | **100** | **87.5**
 :red_circle: | app/client/src/utils/hooks/useProceedToNextTourStep.tsx | 100 **(0)** | 40 **(-10)** | 100 **(0)** | 100 **(0)**</details>

### CODE REVIEW

File: app/client/src/comments/CommentCard/CommentCard.tsx [hunk 1]
```diff
@@ -263,6 +265,11 @@ function CommentCard({
   inline?: boolean;
   visible?: boolean;
 }) {
+  const proceedToNextTourStep = useProceedToNextTourStep(
+    TourType.COMMENTS_TOUR_EDIT_MODE,
+    3,
```
Comment (pranavkanade) [inline app/client/src/comments/CommentCard/CommentCard.tsx:270]: Can we please declare enums for tour step indexes? Will be helpful down the line.

File: app/client/src/reducers/uiReducers/commentsReducer/handleNewCommentThreadEvent.ts [hunk 2]
```diff
@@ -37,6 +37,10 @@ const handleNewCommentThreadEvent = (
 
   const showUnreadIndicator = !state.isCommentMode;
 
+  if (thread.isPrivate) {
+    state.visibleCommentThreadId = thread._id;
+  }
```
Comment (pranavkanade) [inline app/client/src/reducers/uiReducers/commentsReducer/handleNewCommentThreadEvent.ts:42]: QQ: what is the purpose of `isPrivate` property?

**Review by pranavkanade — APPROVED** [review]
One small suggestion and a question. Rest frontend changes LMGT!

File: app/client/src/reducers/uiReducers/commentsReducer/handleNewCommentThreadEvent.ts (hunk 2 above)
Comment (rishabhsaxena) [reply app/client/src/reducers/uiReducers/commentsReducer/handleNewCommentThreadEvent.ts:42]: Currently this is used during onboarding, these comment threads are private to a user. They are triggered from the backend during the tour and we need to switch to the incoming thread. So we update the visible thread id. Would add a comment here as well.

File: app/client/src/comments/CommentCard/CommentCard.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/comments/CommentCard/CommentCard.tsx:270]: Yeah makes sense

**Review by pranavkanade — APPROVED** [review]

**Comment by rishabhsaxena** [comment]
Closing this and cherry-picking changes to another branch

**Pull request closed without merging**